java-lib/src/main/java/com/wavefront/ingester/GraphiteDecoder.java

```diff
@@ -26,19 +27,24 @@
       .appendOptionalTimestamp().whiteSpace()
       .appendAnnotationsConsumer().whiteSpace().build();
   private final String hostName;
+  private LinkedHashSet<String> customSourceTags;
 
-  public GraphiteDecoder() {
+  public GraphiteDecoder(LinkedHashSet<String> customSourceTags) {
```

Comment:
Ah, I see, you need ordering, it's probably better to accept a Set<String> here and pass in an ordered set for use cases that need ordering